internal/storage/inmem/store.go

```diff
@@ -271,6 +279,13 @@ func (s *Store) OwnerReferences(id *pbresource.ID) ([]*pbresource.ID, error) {
 	return refs, nil
 }
 
+func (s *Store) txn(write bool) *memdb.Txn {
+	s.mu.RLock()
```

Comment:
Good call out! I think it's slightly tricky to know what the interface of this method should be, because the lifetimes of the transaction and event lock are different.